source/server/overload_manager_impl.cc

```diff
@@ -223,12 +278,14 @@ void OverloadManagerImpl::updateResourcePressure(const std::string& resource, do
                   const std::string& action = entry.second;
                   auto action_it = actions_.find(action);
                   ASSERT(action_it != actions_.end());
+                  const OverloadActionState old_state = action_it->second.getState();
                   if (action_it->second.updateResourcePressure(resource, pressure)) {
-                    const bool is_active = action_it->second.isActive();
-                    const auto state =
-                        is_active ? OverloadActionState::Active : OverloadActionState::Inactive;
-                    ENVOY_LOG(info, "Overload action {} became {}", action,
-                              is_active ? "active" : "inactive");
+                    const auto state = action_it->second.getState();
```

Comment:
Is additional work needed to optimize update propagation?  It seems that each action is propagated independently to workers.  updateResourcePressure returns true a lot more often than before.  Is there some potential for aggregation?

How often does updateResourcePressure run anyway?  Should the code below wait for propagation to complete before returning as a way to ensure that at most 1 update is in progress?

Reply:
Yea, I think it should be straightforward to aggregate a bunch of <action, state> pairs in this loop and then send them to workers with a single call to runOnAllThreads. Maybe best to do this in a follow-on PR?

Re: how often this is called - our best practices doc for use as an edge proxy has an example config where we set this to 250ms (https://www.envoyproxy.io/docs/envoy/latest/configuration/best_practices/edge)

Reply:
Yeah I think there are a couple optimizations we can apply here. Right now we post an event to each worker for each resource update, but we can do a better job of coalescing those. updateResourcePressure returns true more often only if the proxy is configured with a range trigger, which none should be yet.

Reply:
I'm going to address this an Elisha's other comment about the 'inactive' state in a follow-on PR.